api/v1/services/CheckInService.js

```diff
@@ -0,0 +1,120 @@
+var CheckitError = require('checkit').Error;
+var _Promise = require('bluebird');
+var _ = require('lodash');
+
+var CheckIn = require('../models/CheckIn');
+var UserService = require('../services/UserService');
+var errors = require('../errors');
+var utils = require('../utils');
+
+
+/**
+ * Helper method to deal with boolean flags that should throw errors if requested to be set
+ * to true if already true
```

Comment:
Can you rewrite this? Use a `@throws` in the list of things below to describe errors.